src/Data/Singletons/Single/Monad.hs

```diff
@@ -228,3 +228,6 @@ singDecsM :: DsMonad q => [Dec] -> SgM [DDec] -> q [DDec]
 singDecsM locals thing = do
   (decs1, decs2) <- singM locals thing
   return $ decs1 ++ decs2
+
+proxyFor :: DType -> DExp
```

Comment:
The one use of this function will be able to removed once goldfirere/th-desugar#47 is merged, I think. I don't love the one-off use of the `Proxy` names here, but it should be temporary enough.